extension/page-detect.js

```diff
@@ -33,6 +33,8 @@ window.pageDetect = (() => {
 
 	const isBlame = () => isRepo() && /^\/blame\//.test(getRepoPath());
 
+	const isSingleFile = () => isRepo() && /\/blob\//.test(getRepoPath());
```

Comment:
That seems like an edge case (a folder named `blob`?!), but smart thinking.

I agree on using 1, it also makes sense that all page detection is in one place.